Mathlib/Data/Int/Units.lean

```diff
@@ -34,6 +34,18 @@ theorem isUnit_eq_one_or {a : ℤ} : IsUnit a → a = 1 ∨ a = -1
   | ⟨_, hx⟩ => hx ▸ (units_eq_one_or _).imp (congr_arg Units.val) (congr_arg Units.val)
 #align int.is_unit_eq_one_or Int.isUnit_eq_one_or
 
+@[simp]
+theorem units_ne_neg_self (u : ℤˣ) : u ≠ -u := by
+  rcases units_eq_one_or u with rfl | rfl <;> decide
+
+@[simp]
+theorem neg_units_ne_self (u : ℤˣ) : -u ≠ u := (units_ne_neg_self u).symm
+
+theorem units_ne_iff_eq_neg {u u' : ℤˣ} : u ≠ u' ↔ u = -u' := by
```

Comment:
Can you add `iff`-versions of these lemmas. I think the `iff` version of this lemma can be used to golf `isUnit_eq_or_eq_neg` below.

Reply:
I don't understand what you mean. This one already is an `iff`, and the others don't have an obvious `iff` version.

Reply:
Ah sorry, I meant to say `IsUnit`-versions.

Reply:
Something like this: #6952